### DISCUSSION TITLE
Windows Terminal Preview v1.21.1772.0

### DISCUSSION THREAD

**Discussion by DHowett** [microsoft/terminal#17479] [Announcements]
Like its smaller and more stable cousin, this Preview channel servicing update fixes a handful of issues, including that annoying hyperlink bug. This update is currently rolling out via the store to everyone.

## Changes

* The "built-in box-drawing glyphs" feature (as well as undercurls) now works with Direct2D as well (#17278) (#17454) (#17458) (#17464)

## Bug Fixes

* New windows should more reliably come to the front when Terminal is already running (#17368) (thanks @flyingcat!)
* Autodetected URLs will no longer erroneously include the character after them (#17326) (thanks @tusharsnx!)
* Buffer restoration will no longer destroy the last character on the screen (#17374)
* Search result highlighting no longer snaps incorrectly, acts strange when you open/close the search panel too quickly, or leaves marks on the scrollbar when you close the search panel (#17352) (thanks @tusharsnx!)
* Command history-based suggestions will no longer include deleted characters as whitespace or the entire current command (#17245)

### Reliability

* We fixed a number of crashes that happen when you are closing panes (#17358) (#17333) (#17450)
* Instead of crashing, we will now warn you when you are using a version of MacType that is _known to crash_ Terminal and other software on Windows (#17369) (#17380)
* We've removed one debug assert that is technically possible to encounter during search navigation (#17361)

### Localization

* We have fixed a number of localization issues in Korean, Italian, and Spanish (#17304)

---

_Binary files inside the unpackaged distribution archive bear the version number `1.21.240625002-preview`._

### Asset Hashes

- Microsoft.WindowsTerminalPreview_1.21.1772.0_8wekyb3d8bbwe.msixbundle
   - SHA256 `8F21DE7925DEF91C5BB1C5965544CD10D2A2233E9641D6CD8139C3825DD31CBE`
- Microsoft.WindowsTerminalPreview_1.21.1772.0_arm64.zip
   - SHA256 `1E6849AF93D9D2CF9BA9F61B3855E14D53040E74223010BEC59605560806A785`
- Microsoft.WindowsTerminalPreview_1.21.1772.0_x64.zip
   - SHA256 `6AA37175E2B09170829A39DAF3357D4B88A3965C3C529A45B1B0781B8F3425F0`
- Microsoft.WindowsTerminalPreview_1.21.1772.0_x86.zip
   - SHA256 `95AA80FBEA9BAFB5E91E0B521F7582E56EC57C451B6ADA921A375EF5A1243E4B`

<hr /><em>This discussion was created from the release <a href='https://github.com/microsoft/terminal/releases/tag/v1.21.1772.0'>Windows Terminal Preview v1.21.1772.0</a>.</em>

**Comment by gwgaston** [comment]
The interaction "Remove trailing white-space in rectangular selection" no longer works. White-space is not removed.

**Reply by DHowett** [reply]
Interesting. I can't reproduce this on Canary 1.22 or Preview 1.21. Can you share a copy of your settings file? How are you copying - with Ctrl+C, right-click, or something else?

**Comment by dqqwinner** [comment]
Windows Terminal Preview v1.21.1772.0 cannot automatically create a directory by dragging files to the terminal, which greatly affects work efficiency. Can this function be added? THX

**Reply by DHowett** [reply]
I do not understand. Terminal emulators do not create directories. Can you explain your scenario more?
